blockstore/splitstore/splitstore_compact.go

```diff
@@ -695,17 +703,42 @@ func (s *SplitStore) walkChain(ts *types.TipSet, inclState, inclMsgs abi.ChainEp
 		// the walk is BFS, so we can reset the walked set in every iteration and avoid building up
 		// a set that contains all blocks (1M epochs -> 5M blocks -> 200MB worth of memory and growing
 		// over time)
-		walked = cid.NewSet()
+		walked = newConcurrentVisitor()
 		walking := toWalk
 		toWalk = nil
```

Comment:
Technically we don't need this copy; `walking` could be set to nil after populating `workch`

Reply:
ah yes, and we can reuse the slice; let me fix.

Reply:
hrm, we can't actually reuse the slice, it first comes from the tipset; will just setr to nil after copying.

Reply:
done.

Reply:
you know, we can first make a new one and copy when we initialize and then reuse; that will save somr allocations.
let me do.

Reply:
done done.